src/app/rpmostree-compose-builtin-tree.c

```diff
@@ -849,107 +956,39 @@ impl_compose_tree (const char      *treefile_pathstr,
-
-  { glnx_unref_object JsonGenerator *generator = json_generator_new ();
-    char *treefile_buf = NULL;
-    gsize len;
-
-    json_generator_set_root (generator, treefile_rootval);
-    json_generator_set_pretty (generator, TRUE);
-    treefile_buf = json_generator_to_data (generator, &len);
-
-    self->serialized_treefile = g_bytes_new_take (treefile_buf, len);
-  }
-
-  /* Download rpm-md repos, packages, do install */
-  g_autofree char *new_inputhash = NULL;
-  { gboolean unmodified = FALSE;
-
-    if (!install_packages_in_root (self, treefile, rootfs_fd,
-                                   (char**)packages->pdata,
-                                   opt_force_nocache ? NULL : &unmodified,
-                                   &new_inputhash,
-                                   cancellable, error))
-      return FALSE;
-
-    if (unmodified)
-      {
-        g_print ("No apparent changes since previous commit; use --force-nocache to override\n");
-        /* Note early return */
-        return TRUE;
-      }
-    else if (opt_dry_run)
-      {
-        g_print ("--dry-run complete");
-        if (opt_touch_if_changed)
-          g_print (", updating --touch-if-changed=%s", opt_touch_if_changed);
-        g_print ("; exiting\n");
-        if (!process_touch_if_changed (error))
-          return FALSE;
-        /* Note early return */
-        return TRUE;
-      }
-  }
-
-  /* Destroy this now so the libdnf stack won't have any references
-   * into the filesystem before we manipulate it.
-   */
-  g_clear_object (&self->corectx);
-
-  if (g_strcmp0 (g_getenv ("RPM_OSTREE_BREAK"), "post-yum") == 0)
-    return FALSE;
-
   /* Start postprocessing */
-  if (!rpmostree_treefile_postprocessing (rootfs_fd, self->treefile_context_dirs->pdata[0],
-                                          self->serialized_treefile, treefile,
+  if (!rpmostree_treefile_postprocessing (self->rootfs_dfd, self->treefile_context_dirs->pdata[0],
+                                          self->serialized_treefile, self->treefile,
                                           next_version, cancellable, error))
     return glnx_prefix_error (error, "Postprocessing");
 
-  if (!rpmostree_prepare_rootfs_for_commit (self->workdir_dfd, &rootfs_fd, rootfs_name,
-                                            treefile,
+  const char rootfs_name[] = "rootfs.tmp";
+  if (!rpmostree_prepare_rootfs_for_commit (self->workdir_dfd, &self->rootfs_dfd, rootfs_name,
+                                            self->treefile,
                                             cancellable, error))
     return glnx_prefix_error (error, "Preparing rootfs for commit");
 
-  g_autoptr(GFile) treefile_dirpath = g_file_get_parent (self->treefile);
-  if (!rpmostree_check_passwd (self->repo, rootfs_fd, treefile_dirpath, treefile,
+  g_autoptr(GFile) treefile_dirpath = g_file_get_parent (self->treefile_path);
+  if (!rpmostree_check_passwd (self->repo, self->rootfs_dfd, treefile_dirpath, self->treefile,
```

Comment:
Hey not sure if this will happen. It seems like when impl_commit_tree is called only via `rpm-ostree compose tree --commit-root`, the self->previous_checksum will be NULL as the initialization only happens in 
`
if (!ostree_repo_read_commit (self->repo, self->ref, &self->previous_root, &self->previous_checksum,                           cancellable, &temp_error))
`.

And the above line seems to come from only in impl_install_tree()?  Sorry if I made a mistake, just a bit curious :p

Reply:
Hm, yep, I think you're right.  I'm going to rework this patch down the line so `rpm-ostree compose commit-root` is really close to `rpmostree_final_postprocess() + rpmostree_commit()`.  So we won't do anything like the passwd checking stuff.